src/main/java/github/scarsz/discordsrv/DiscordSRV.java

```diff
@@ -1879,53 +1877,66 @@ public static Message translateMessage(MessageFormat messageFormat, BiFunction<S
+        OfflinePlayer player = null;
+        if (username != null && offline) {
+            // resolve username to player/uuid
+            //TODO resolve name to online uuid when offline player is present
+            // (can't do it by calling Bukkit.getOfflinePlayer(username).getUniqueId() because bukkit just returns the offline-mode CraftPlayer)
+            player = Bukkit.getOfflinePlayer(username);
+            uuid = player.getUniqueId();
+            offline = PlayerUtil.uuidIsOffline(uuid);
+        }
+        if (username == null && uuid != null) {
+            // resolve uuid to player/username
+            player = Bukkit.getOfflinePlayer(uuid);
+            username = player.getName();
+        }
+        if (StringUtils.isBlank(texture) && player != null && player.isOnline()) {
+            // grab texture placeholder from player if online
+            texture = NMSUtil.getTexture(player.getPlayer());
+        }
+
         String avatarUrl = DiscordSRV.config().getString("AvatarUrl");
-        if (StringUtils.isBlank(avatarUrl)) avatarUrl = "https://crafatar.com/avatars/{uuid-nodashes}?overlay&size={size}";
-        if (profile != null) {
-            avatarUrl = avatarUrl
-                    .replace("{texture}", profile.getSkin())
-                    .replace("{username}", profile.getName())
-                    .replace("{uuid}", profile.getUuid().toString())
-                    .replace("{uuid-nodashes}", profile.getUuid().toString().replace("-", ""))
-                    .replace("{size}", "128");
-            avatarUrl = PlaceholderUtil.replacePlaceholdersToDiscord(avatarUrl, Bukkit.getOfflinePlayer(profile.getUuid()));
-        } else {
-            avatarUrl = avatarUrl
-                    .replace("{texture}", "")
-                    .replace("{username}", username)
-                    .replace("{uuid}", "c06f8906-4c8a-4911-9c29-ea1dbd1aab82")
-                    .replace("{uuid-nodashes}", "c06f8906-4c8a-4911-9c29-ea1dbd1aab82".replace("-", ""))
-                    .replace("{size}", "128");
-            avatarUrl = PlaceholderUtil.replacePlaceholdersToDiscord(avatarUrl);
+        if (StringUtils.isBlank(avatarUrl)) {
+            avatarUrl = !offline ? "https://crafatar.com/avatars/{uuid-nodashes}.png?size={size}&overlay#{texture}"
+                                 : "https://minotar.net/helm/{username}/{size}.png#{texture}";
+        }
+        if (offline && !avatarUrl.contains("{username}")) {
+            if (!offlineUuidAvatarUrlNagged) {
+                DiscordSRV.error("Your AvatarUrl does not contain the {username} placeholder even though this server is using offline UUIDs.");
+                DiscordSRV.error("You should set your AvatarUrl to https://minotar.net/helm/{username}/{size}.png#{texture} to get avatars to work.");
+                offlineUuidAvatarUrlNagged = true;
+            }
         }
 
-        debug("Avatar URL generated: " + avatarUrl);
```

Comment:
Could still be useful to log this

Reply:
Opted not to because of the frequency that they would be shown, I'm down to add it when granular debug options are a thing

Reply:
Fair point